### ISSUE TITLE
Dice score for test data

### ISSUE THREAD

**Issue by sivajyothi1** [MIC-DKFZ/nnUNet#65]
Hi @FabianIsensee , Thank you very much for sharing your wonderful work.
nnUnet worked well for training my own data. 

Can you please clarify my doubt..
How to get the DICE score for the test data, comparing ground truth and the predicted results?
How to evaluate the model.

**Comment by FabianIsensee** [comment]
Hi,
nnU-Net typically does not have the test data and the evaluation is done externally through challenge platforms. It is cleanest if you have your test data very far away from the rest of your training data :-)
But still, if you wish to evaluate predictions, you can use this function https://github.com/MIC-DKFZ/nnUNet/blob/648f4b3fb9fa7f8e365db6793248e461f2e7331c/nnunet/evaluation/evaluator.py#L448
Best,
Fabian

**Comment by sivajyothi1** [comment]
Hi @FabianIsensee, Thank you for replying.
As i am new to deep learning and did not understand how to evaluate predictions for test data.
you mentioned me to use evaluate_ folder in evaluator.py,
But i didn't understand that

-  for folder_with_gts  and folder_with_predictions in evaluate_folder, should i mention folder path where the ground truth images and prediction images are located??
- how to mention labels?

Can you please explain me how to use the evaluate_folder for evaluation?
It will be great help for me.

Thank you.

**Comment by FabianIsensee** [comment]
Hi,
like the name implies
folder_with_gts is the folder where the ground truth nitfi labels are saved
folder_with_predictions is the folder where the predicted nitfi labels are saved
labels are the labels in the dataset (0, 1, 2, 3) for BraTS for example
Best,
Fabian

**Comment by sivajyothi1** [comment]
Hello @FabianIsensee ,
It's a great work. Thank you very much. I got my dice score.
Best,
Siva.

**Comment by linden1994** [comment]
> Hello @FabianIsensee ,
> It's a great work. Thank you very much. I got my dice score.
> Best,
> Siva.
How to get dice score，i run the  evaluator.py ，but don‘t work，Looking forward to your reply

**Comment by ironb25** [comment]
Hi @Fabian

Hi Fabian,

I have ran training 5 fold on BRATS 2018 dataset, so have 5 fold models.And want to evaluate on our test set of 30 patients.I would like to get dice scores of each of 3 classes(edema,NET, enhancing tumoor) for this test set.
alsoo I did ran data integrity on my task for the dataset, which ran fine.

This is what i did.

1) nnUNet_find_best_configuration -t 1 -m 3d_fullres

  

2) nnUNet_predict -i /mnt/rstor/CSE_BME_AXM788/home/rxb362/projects/segmentation/nnunet_dl/nnUNet_raw/nnUNet_raw_data/Task110_MultiInsti_GBM_Upenn/imagesTr/ -o /mnt/rstor/CSE_BME_AXM788/home/rxb362/projects/segmentation/nnunet_dl/nnUnet_inference -tr nnUNetTrainerV2 -ctr nnUNetTrainerV2CascadeFullRes -m 3d_fullres -p nnUNetPlansv2.1 -t Task001_BrainTumour

 

I got a post processing.json in the Task110 folder but it seemed to be same file as the one in Task001 after I ran nnUNet_find_best_configuration.



So I looked upon the repo and found this https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/common_questions.md#evaluating-test-set-results

 

3) nnUNet_evaluate_folder -ref /mnt/rstor/CSE_BME_AXM788/home/rxb362/projects/segmentation/nnunet_dl/nnUNet_raw/nnUNet_raw_data/Task110_MultiInsti_GBM_Upenn/labelsTr/ -pred /mnt/rstor/CSE_BME_AXM788/home/rxb362/projects/segmentation/nnunet_dl/nnUnet_inference -l 1 2 3 



This generated a summary.json but the dice scores of 3 classes look very bad.

Not sure if I did something wrong? I have uploaded the postprocessing.json which is  from BRATS 5 fold model generated while running nnUnet_predict, and summary.json generated while running evaluate.
[Summary_Postprocessing.zip](https://github.com/MIC-DKFZ/nnUNet/files/5926141/Summary_Postprocessing.zip)







Thanks,

Rohan

**Comment by FabianIsensee** [comment]
What you did is correct. It seems like your test data is from a different distribution, the test labels are defined differently (1 is not edema or something similar) OR the order of the modalities is not correct. Visually inspect your test images to find the error, it is very likely that it is just an oversighton your end. Also please use the [Discussion](https://github.com/MIC-DKFZ/nnUNet/discussions) for this. You are lucky I saw your post in this closed issue ;-)
